Source/WebCore/platform/mediastream/gstreamer/DesktopPortal.cpp

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright (C) 2025 Igalia S.L
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * aint with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#include "config.h"
+#include "DesktopPortal.h"
+
+#if ENABLE(MEDIA_STREAM) && USE(GSTREAMER)
+
+#include "GRefPtrGStreamer.h"
+#include <gio/gunixfdlist.h>
+#include <gst/video/video-format.h>
+#include <optional>
+#include <unistd.h>
+#include <wtf/Scope.h>
+#include <wtf/UUID.h>
+#include <wtf/WeakRandomNumber.h>
+#include <wtf/glib/GUniquePtr.h>
+
+#if GST_CHECK_VERSION(1, 24, 0)
+#include <gst/video/video-info-dma.h>
+#endif
+
+namespace WebCore {
+
+static const Seconds s_dbusCallTimeout = 10_ms;
+
+GRefPtr<GDBusProxy> createDBusProxy(const String& interfaceName)
```

Comment:
As far as I can see there's no reason for this to be a `String`, you can use an `ASCIILIteral` for less overhead.